src/plugins/data/common/data_views/data_views/data_view.ts

```diff
@@ -391,24 +425,38 @@ export class DataView implements IDataView {
     );
   }
 
-  /**
-   * Converts a DataView to a serializable Dataset object suitable for storage in Redux
-   * Maps dataSource to dataSourceRef and includes only essential properties
-   * TODO: This should be more complete
-   */
-  public toDataset(): Dataset {
+  public async toDataset(): Promise<Dataset> {
+    const defaultType = DEFAULT_DATA.SET_TYPES.INDEX_PATTERN;
+    const dataSourceReference = this.dataSourceRef || (this as any).dataSource;
+
+    let dataSource;
+    if (dataSourceReference?.id) {
+      try {
+        const dataSourceSavedObject = await this.savedObjectsClient.get(
+          dataSourceReference.type,
+          dataSourceReference.id
+        );
+        const attributes = dataSourceSavedObject.attributes as any;
+        dataSource = {
+          id: dataSourceReference.id,
+          title: attributes.title || dataSourceReference.id,
+          type: attributes.dataSourceEngineType || dataSourceReference.type || 'OpenSearch',
```

Comment:
`'OpenSearch'` should come from the constant?

Reply:
Should it even fallback to 'OpenSearch'? I feel it should just throw error